src/ui/pages/Backup/useBackupContext.ts

```diff
@@ -0,0 +1,34 @@
+import { invariant } from 'src/shared/invariant';
+import { useSearchParams } from 'react-router-dom';
+import { urlContext } from 'src/ui/shared/UrlContext';
+
+export type BackupContext =
+  | {
+      appMode: 'onboarding';
+    }
+  | {
+      appMode: 'wallet';
+      groupId: string;
+    };
+
+/**
+ * Combines URL search parameters from the window's location and react-router's search params.
+ * @returns {URLSearchParams} Merged search parameters from both sources.
+ */
+function useUrlParams(): URLSearchParams {
+  const params = new URLSearchParams(window.location.search);
+  const [hashParams] = useSearchParams();
+  return new URLSearchParams([...params.entries(), ...hashParams.entries()]);
+}
+
+export function useBackupContext(): BackupContext {
```

Comment:
I don't see why we need this hook

Consumers can read `appMode` from `urlContext` and `groupId` from `useSearchParams`

Reply:
It allows not to do `invariant(groupId, "error msg")` every time

Reply:
Ok, lets remove it

Reply:
✅ removed